code/game/objects/items/melee/misc.dm

```diff
@@ -677,55 +677,50 @@
 	righthand_file = 'icons/mob/inhands/weapons/melee_righthand.dmi'
 	slot_flags = ITEM_SLOT_BELT
-	force = 0.001
+	force = 0.001 //"Some attack noises shit"
 	reach = 3
 	w_class = WEIGHT_CLASS_NORMAL
 	attack_verb = list("flogged", "whipped", "lashed", "disciplined")
 	hitsound = 'sound/weapons/whip.ogg'
 
 /obj/item/melee/curator_whip/attack(mob/living/target, mob/living/user)
 	. = ..()
-	var/mob/living/carbon/human/H = target
-	var/mob/living/carbon/human/U = user
-	if(ishuman(target))
-		if((user.zone_selected == BODY_ZONE_CHEST) || (user.zone_selected == BODY_ZONE_HEAD) || (user.zone_selected == BODY_ZONE_PRECISE_GROIN))
-			if(H.getarmor(type = "melee") < 16)
-				H.emote("scream")
-				H.visible_message("<span class='danger'>[U] whips [H]!</span>", "<span class='userdanger'>[U] whips you! It stings!</span>")
-		if((user.zone_selected == BODY_ZONE_R_LEG) || (user.zone_selected == BODY_ZONE_L_LEG))
-			var/dist = get_dist(H, U)
-			if(dist < 2)
-				to_chat(user, "<span class='warning'>[H] is too close to trip with the whip!</span>")
-				return
-			else
-				target.Knockdown(30)
-				log_combat(user, target, "tripped", src)
-				H.visible_message("<span class='danger'>[U] trips [H]!</span>", "<span class='userdanger'>[U] whips your legs out from under you!</span>")
-			return
-		if(user.zone_selected == BODY_ZONE_L_ARM)
-			var/obj/item/I = H.get_held_items_for_side("left")
-			if(I)
-				if(H.dropItemToGround(I))
-					H.visible_message("<span class='danger'>[I] is yanked off [H]'s hand by [src]!</span>","<span class='userdanger'>[U] grabs [I] with [src]!</span>")
-					if(!user.get_inactive_held_item())
-						U.throw_mode_on()
-						U.swap_hand()
-						I.throw_at(user, 10, 2)
-						to_chat(user, "<span class='notice'>You pull [I] towards yourself.</span>")
-			log_combat(user, target, "disarmed", src)
-			H.visible_message("<span class='danger'>[U] disarms [H]!</span>", "<span class='userdanger'>[U] disarmed you!</span>")
-		if(user.zone_selected == BODY_ZONE_R_ARM)
-			var/obj/item/I = H.get_held_items_for_side("right")
-			if(I)
-				if(H.dropItemToGround(I))
-					H.visible_message("<span class='danger'>[I] is yanked off [H]'s hand by [src]!</span>","<span class='userdanger'>[U] grabs [I] with [src]!</span>")
-					to_chat(user, "<span class='notice'>You pull [I] towards yourself.</span>")
-					if(!user.get_inactive_held_item())
-						U.throw_mode_on()
-						U.swap_hand()
-						I.throw_at(user, 10, 2)
+	if(!ishuman(target))
+		return
+
+	switch(user.zone_selected)
+		if(BODY_ZONE_L_ARM)
+			whip_disarm(user, target, "left")
+		if(BODY_ZONE_R_ARM)
+			whip_disarm(user, target, "right")
+		if(BODY_ZONE_R_LEG)
+			whip_trip(user, target)
+		if(BODY_ZONE_L_LEG)
+			whip_trip(user, target)
+		else
+			if(target.getarmor(type = "melee") < 16)
+				target.emote("scream")
+				target.visible_message("<span class='danger'>[user] whips [target]!</span>", "<span class='userdanger'>[user] whips you! It stings!</span>")
```

Comment:
shouldn't you always be checking for armor? Do you need that else here?

Reply:
I assumed the intent was "if theyre wearing armor they wont feel the lashes" or some shit.

The else is to cover body parts that arent given an explicit condition in the switch. thats kind of an important thing.